solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -241,6 +241,7 @@ private HttpClient createHttpClient(Builder builder) {
     try {
       httpClient.start();
     } catch (Exception e) {
+      close(); // make sure we clean up
```

Comment:
Make sure we cleanup on a failed start. At least the executor could have been created at this point. Other things might have been created as well.